src/main/java/com/minecraftabnormals/abnormals_core/core/events/CompatEvents.java

```diff
@@ -59,4 +77,26 @@ public static void onUpdateEntity(LivingEvent.LivingUpdateEvent event) {
 			if (entity.getPersistentData().getBoolean(POISON_TAG)) ((IAgeableEntity) entity).resetGrowthProgress();
 		}
 	}
+
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public static void onNoteBlockPlay(NoteBlockEvent.Play event) {
+		if (SORTED_CUSTOM_NOTE_BLOCK_INSTRUMENTS  != null) {
+			IWorld world = event.getWorld();
+			BlockPos pos = event.getPos();
+			BlockState state = world.getBlockState(pos.relative(Direction.DOWN));
+			for (DataUtil.CustomNoteBlockInstrument instrument : SORTED_CUSTOM_NOTE_BLOCK_INSTRUMENTS) {
+				if (instrument.test(state)) {
+					SoundEvent sound = instrument.getSound();
+					double note = event.getVanillaNoteId();
+					world.playSound(null, pos, sound, SoundCategory.RECORDS, 3.0F, (float) Math.pow(2.0D, (note - 12) / 12.0D));
+					ResourceLocation noteId = ForgeRegistries.PARTICLE_TYPES.getKey(ParticleTypes.NOTE);
```

Comment:
It'd be better to make the `ResourceLocation` for the particle type a constant to make it not get looked up every time.

Reply:
Done.